internal/app/networkd/pkg/address/dhcp.go

```diff
@@ -37,7 +37,8 @@ func (d *DHCP) Link() *net.Interface {
 }
 
 // Discover handles the DHCP client exchange stores the DHCP Ack.
-func (d *DHCP) Discover(ctx context.Context) error {
+func (d *DHCP) Discover(ctx context.Context, link *net.Interface) error {
+	d.NetIf = link
 	// TODO do something with context
 	ack, err := d.discover()
 	d.Ack = ack
```

Comment:
Is this all thread safe? There is a one-to-one match for a `DHCP` to a `*net.Interface`?

Reply:
I _think_ so. Link has a bunch of useful information, but I think at the end of the day all I care about is the index. I think I'd be more inclined to follow up later with another PR to potentially reduce this scope.